tests/test_session.py

```diff
@@ -132,3 +131,51 @@ async def test_session_requests(
     assert requests[2]["url"].endswith("/v0.6/stats?test_session_token=test_case")
     assert requests[3]["method"] == "POST"
     assert requests[3]["url"].endswith("/telemetry/proxy/api/v2/apmtelemetry?test_session_token=test_case")
+
+
+async def test_404_when_session_doesnt_exist(agent):
+    """When a session that doesn't exist is requested, we should get an error."""
+    resp = await agent.get("/test/session/traces", params={"test_session_token": "nonexistent"})
+    assert resp.status == 404
+
+
+async def test_empty_session_ok(agent):
+    """When a session exists but has no traces, we should get an OK with an empty list returned"""
+    resp = await agent.get("/test/session/start", params={"test_session_token": "emptysession"})
+    assert resp.status == 200
+    resp = await agent.get("/test/session/traces", params={"test_session_token": "emptysession"})
+    assert resp.status == 200
+    assert await resp.json() == []
+
+
+async def test_session_association_of_untokenized_traces(agent, do_reference_v04_http_trace):
+    """Requests sent without a session token are associated with the session that proceeds it"""
+    resp = await agent.get("/test/session/start", params={"test_session_token": "sessiona"})
+    assert resp.status == 200
+    await do_reference_v04_http_trace()
+    resp = await agent.get("/test/session/start", params={"test_session_token": "sessionb"})
+    assert resp.status == 200
+    resp_a = await agent.get("/test/session/traces", params={"test_session_token": "sessiona"})
+    assert resp_a.status == 200
+    resp_b = await agent.get("/test/session/traces", params={"test_session_token": "sessionb"})
+    assert resp_b.status == 200
+    assert len(await resp_a.json()) == 1
+    assert await resp_b.json() == []
+
+    await do_reference_v04_http_trace()
+    resp_a = await agent.get("/test/session/traces", params={"test_session_token": "sessiona"})
+    assert resp_a.status == 200
+    resp_b = await agent.get("/test/session/traces", params={"test_session_token": "sessionb"})
+    assert resp_b.status == 200
+    assert len(await resp_a.json()) == 1
+    assert len(await resp_b.json()) == 1
```

Comment:
Can we extend this just a tad further to see that if we sent more traces under a _new_ `sessiona` that those traces are returned rather than from the older `sessiona` session?